### ISSUE TITLE
[Failed] Twitter

### ISSUE THREAD

**Issue by soimort** [soimort/you-get#1139]
The result returned by `you-get` is a M3U8 file, not the actual video.

```
$ you-get -di https://twitter.com/take_miyu112/status/731386804792778752
[DEBUG] get_content: https://twitter.com/i/videos/731386804792778752?embed_source=facebook
Site:       Twitter.com
Title:      take_miyu112 [731386804792778752]
Type:       Unknown type (None)
Size:       0.0 MiB (512 Bytes)
```

**Comment by cnbeining** [comment]
How can we use ```VideoExtractor``` with pics?

I just generated a pair of key and secret to take advantage of their API.

**Comment by soimort** [comment]
> How can we use VideoExtractor with pics?
> 
> I just generated a pair of key and secret to take advantage of their API.

What are you talking about?

**Comment by cnbeining** [comment]
I am talking about using Twitter's API instead of parsing the webpage.

They are offering different resolutions of video; it should be wise to use the API.

**Comment by soimort** [comment]
@cnbeining I believe Twitter's developer API key is not something we're gonna use in this (public) project, as long as it involves your own secret and takes up your personal quota.

It would be good to have a completely key-free API like [Tumblr API v1](https://soimort.tumblr.com/api/read); while Twitter doesn't have anything like that.

P.S. This sadly reminds me of your gone Bilibili key...

**Comment by cnbeining** [comment]
For some reason **other than** abuse that key was banned. Sigh.

I will look through the documentation, maybe exchange a few emails with Twitter...

**Comment by soimort** [comment]
And perhaps their [TOS](https://dev.twitter.com/overview/terms/policy) as well:

tl;dr: You need to keep your own API keys or other access credentials private.

For the use case of `you-get`, you can't both keep an API token private and make use of it at the same time.
